### PR TITLE
feat: add various RE files for Starfield

### PR DESCRIPTION
**Pull request by gottyduke**
This PR brings all of current SFSE RE files in sync with CommonLibSF.
Most of the RE files are tested, I couldn't test BSScript related stuff because currently don't seem like a viable way to do so.

CI will fail for clang-cl because there's an unreferenced parameter in `BSTEventSink::ProcessEvent`, there's no way in hell I'm going to that with a `[[maybe_unused]]`, because that function is meant to be overriden by all sub events and that parameter will never be unreferenced.

I suggest rebasing this PR or Merging instead of Squash, the sheer amount of files this PR brings if squashed into one commit, will make future diff checks difficult, atleast for human readable means. Scrolling through a 100+ files commit to find the change is painful.

### CODE REVIEW

**Comment by shadeMe** [comment]
> Most of the RE files are tested, I couldn't test BSScript related stuff because currently don't seem like a viable way to do so.

Tested in which sense?

**Comment by ThirdEyeSqueegee** [comment]
Possibly in violation of SFSE license (or lack thereof)? We should maybe ask Ian about this

**Comment by gottyduke** [comment]
> Tested in which sense?

By hooking a runtime game and browse structures in x64dbg.

**Comment by gottyduke** [comment]
> Possibly in violation of SFSE license (or lack thereof)? We should maybe ask Ian about this

https://github.com/Starfield-Reverse-Engineering/CommonLibSF/pull/4#issuecomment-1707729867

**Review by powerof3 — COMMENTED** [review]
IMO this should be `BGS` or `SF`, this isn't a TES game (a non-issue tho)
BSReflection types could be in a single header (having headers like IObject.h might be confusing when other similar named classes exist like `BSScript::IObjectPolicy`)

**Comment by ThirdEyeSqueegee** [comment]
> i think it's fine, but it's a little disingenious to be literally copy pasting and changing it for purely style purposes and not actually doing any RE.
i think it's kind of pointless to push classes that are just straight wrong or havent even been remotely verified

From Expired. I'm inclined to agree, we should prolly only intake non-empty and tested classes

**Comment by powerof3** [comment]
> > i think it's fine, but it's a little disingenious to be literally copy pasting and changing it for purely style purposes and not actually doing any RE.
> > i think it's kind of pointless to push classes that are just straight wrong or havent even been remotely verified
> 
> From Expired. I'm inclined to agree, we should prolly only intake non-empty and tested classes

Even if they're empty they serve as a skeleton for further RE work, and any errors or mistakes can be corrected when found. And  given that Expired's own work consists of the RE I did, it's disingenuous to say we're just copy pasting from SFSE.

**Comment by ThirdEyeSqueegee** [comment]
Alrighty I'm good as long as everyone knows where everyone else stands, ping me whenever this is ready to merge and I'll rebase it

File: CommonLibSF/include/RE/T/TESAIForm.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		std::uint16_t unk00;           // 00
+		std::uint8_t  unk02;           // 02
+		std::uint8_t  pad03;           // 03
+		std::uint16_t aggroRadius[4];  // 04
+		std::uint16_t pad0A;           // 0A
+		std::uint8_t  unk0C;           // 0C
+	};
+	static_assert(sizeof(AIDATA_GAME) == 0x10);
+
+	class TESAIForm : public BaseFormComponent
```
Comment (FlayaN) [inline CommonLibSF/include/RE/T/TESAIForm.h:34]: Shouldn't these also be renamed from `TES*` -> `SF*`? Both filename and classname

**Comment by gottyduke** [comment]
> From Expired. I'm inclined to agree, we should prolly only intake non-empty and tested classes

Allow me to clarify, I did not just copy and paste them over with style changes. If you look at the commit log, I was adding them one by one by hand, separating them with past clib conventions and testing, I have been cross referencing x64dbg output and clibf4.

Don't get me wrong: this is a work based on others, and I appreciate what they have done. My addition might be wrong even if I tested them, but it's easy for people to correct them in the future.

File: CommonLibSF/include/RE/T/TESAIForm.h (hunk 1 above)
Comment (gottyduke) [reply CommonLibSF/include/RE/T/TESAIForm.h:34]: No those are exactly named as is, in the RTTI.

**Comment by gottyduke** [comment]
@ThirdEyeSqueegee @powerof3 I think we are ready to merge if this looks okay to everyone.

**Comment by ThirdEyeSqueegee** [comment]
re. the clang-cl build failure, maybe we should just get rid of the `/WX` flag? Don't see why that needs to be passed really

**Comment by gottyduke** [comment]
> re. the clang-cl build failure, maybe we should just get rid of the /WX flag? Don't see why that needs to be passed really

that could be a solution, I suggest looking into it with a more granular approach, in a future PR maybe. For this PR it's unrelated.

**Pull request merged**